File: Javascript Codewars/6kyu/Exclamation marks series #17: Put the exclamation marks and question marks on the balance - are they balanced?.js
/*
Description

Each exclamation mark's weight is 2; each question mark's weight is 3. Putting two strings left and right on the balance - are 
they balanced?

If the left side is more heavy, return "Left"; if the right side is more heavy, return "Right"; if they are balanced, return 
"Balance".

Examples
"!!", "??"     -->  "Right"
"!??", "?!!"   -->  "Left"
"!?!!", "?!?"  -->  "Left"
"!!???!????", "??!!?!!!!!!!"  -->  "Balance"
*/

function balance(left, right) {
  let l = left
    .split("")
    .map((x) => (x === "!" ? (x = 2) : (x = 3)))
    .reduce((a, b) => a + b, 0);
  let r = right
    .split("")
    .map((x) => (x === "!" ? (x = 2) : (x = 3)))
    .reduce((a, b) => a + b, 0);
  return l > r ? "Left" : l === r ? "Balance" : "Right";
}

File: Python Codewars/6kyu/Exclamation marks series #17: Put the exclamation marks and question marks on the balance - are they balanced?.py
'''
Description

Each exclamation mark's weight is 2; each question mark's weight is 3. Putting two strings left and right on the balance - 
are they balanced?

If the left side is more heavy, return "Left"; if the right side is more heavy, return "Right"; if they are balanced, 
return "Balance".

Examples
"!!", "??"     -->  "Right"
"!??", "?!!"   -->  "Left"
"!?!!", "?!?"  -->  "Left"
"!!???!????", "??!!?!!!!!!!"  -->  "Balance"
'''

def balance(left, right):
    left_count = left.count("!")*2 + left.count("?")*3
    right_count = right.count("!")*2 + right.count("?")*3
    
    if (left_count > right_count):
     return "Left"
    elif (right_count>left_count):
     return "Right"
    else:
     return "Balance"